RESILIENCE.md

```diff
@@ -0,0 +1,46 @@
+# Resilience to Partial Outage and Partial Success
+
+When publishing a batch of events with a [Events.publish](https://github.com/zalando-nakadi/kanadi/blob/d77479b5ef4a6837303fe71a9dc625e7bb8c573d/src/main/scala/org/zalando/kanadi/api/EventsInterface.scala#L9) API,
+
+```scala
+def publish[T](name: EventTypeName, events: List[Event[T]], fillMetadata: Boolean = true): Future[Unit]
+```
+
+the publishing could either _succeed_, _partially succeed_ or _fail_. On success the publishing operation returns `Future[Unit]` that resolves to a `Unit` value. On _partial success_ or _failure_ the `Future` fails with an exception:
+
+- `EventValidation(batchItemResponse: List[BatchItemResponse])` - on partial success;
+- `HttpServiceError | GeneralError | OtherError` - on failure;
+
+On partial sucess, `EventValidation` exception contains `List[BatchItemResponse]` with [statuses](https://nakadi.io/manual.html#definition_BatchItemResponse) of submission for each event in the batch.
+
+The Kanadi library contains strategies to handle publishig errors:
+
+## Fail on Partial Success or Error (Default)
+
+The publishing operation immediately fails if the server returns an error that batch was partially successful or failed.
+In this case the application can retry the whole batch or retry only failed events, depending on type of exception.
+The decision is up to the application. However, it should be noted, that the application **should not retry the whole batch without a backoff strategy**, otherwise it can create problems for server, when many clients retry the same batch over and over.
+
+## Retry with Expponential Backoff
```

Comment:
is there a way to retry the whole batch? some configurable retry strategy e.g. via the code.

Reply:
at the moment this feature is absent in the library,
but some internal projects have wrappers on top that implement exactly this functionality.

Obviously it can be added, just stating the current state.

Reply:
👍